test/mpi/dtpools/src/dtpools.c

```diff
@@ -320,88 +333,16 @@ int DTP_obj_create(DTP_t dtp, int obj_idx, int val_start, int val_stride, MPI_Ai
-            case DTPI_OBJ_LAYOUT_SIMPLE__VECTOR:
-            case DTPI_OBJ_LAYOUT_SIMPLE__HVECTOR:
-            case DTPI_OBJ_LAYOUT_SIMPLE__INDEXED:
-            case DTPI_OBJ_LAYOUT_SIMPLE__HINDEXED:
-            case DTPI_OBJ_LAYOUT_SIMPLE__BLOCK_INDEXED:
-            case DTPI_OBJ_LAYOUT_SIMPLE__BLOCK_HINDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK__VECTOR:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK__HVECTOR:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK__INDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK__HINDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK__BLOCK_INDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK__BLOCK_HINDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK__SUBARRAY_C:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK__SUBARRAY_F:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT__VECTOR:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT__HVECTOR:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT__INDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT__HINDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT__BLOCK_INDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT__BLOCK_HINDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT__SUBARRAY_C:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT__SUBARRAY_F:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__VECTOR:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__HVECTOR:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__INDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__HINDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__BLOCK_INDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__BLOCK_HINDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__SUBARRAY_C:
-            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__SUBARRAY_F:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__VECTOR:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__HVECTOR:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__INDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__HINDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__BLOCK_INDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__BLOCK_HINDEXED:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__SUBARRAY_C:
-            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__SUBARRAY_F:
-                err = MPI_Type_free(&dtp->DTP_obj_array[obj_idx].DTP_obj_type);
-                if (err) {
-                    DTPI_Print_error(err);
-                }
-                dtp->DTP_obj_array[obj_idx].DTP_obj_count = 0;
-                free(dtp->DTP_obj_array[obj_idx].DTP_obj_buf);
-                dtp->DTP_obj_array[obj_idx].DTP_obj_buf = NULL;
-                free(dtp->DTP_obj_array[obj_idx].private_info);
-                dtp->DTP_obj_array[obj_idx].private_info = NULL;
-                break;
-            default:
-                err = DTP_ERR_OTHER;
-                fprintf(stdout, "Type structure %d is not defined\n", obj_idx);
-                fflush(stdout);
-        }
+        err = destructors[obj_type] (dtp, obj_idx);
```

Comment:
In the old code there is a mechanism to check whether `obj_index` is within an expected range. Does the new code have a similar logic?

Reply:
I will add the logic here too

Reply:
Actually `obj_idx` does not need to be checked because is retrieved from the type object handler (which was previously verified as correct).